src/auth/guards/magic-link.guard.ts

```diff
@@ -1,8 +1,42 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
-import { Injectable } from '@nestjs/common';
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
 
 @Injectable()
-export class MagicLinkGuard extends AuthGuard('magic-link') {}
+export class MagicLinkGuard extends AuthGuard('magic-link') {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any
+  handleRequest(err: any, user: any, info: any, context: any, status: any) {
```

Comment:
It is lame the passport library doesn't have better types than this, but nothing you can do 🤷

Reply:
`context` should be `ExecutionContext` and `status` is optional. We can update this method to make it stricter

Reply:
You can still override types for `err`, `user` (this needs to be parameterized), and `info` because they are `any`. TS class extension will allow for that